src/main/scala/org/broadinstitute/dsde/firecloud/service/LibraryService.scala

```diff
@@ -83,7 +83,15 @@ class LibraryService (protected val argUserInfo: UserInfo, val rawlsDAO: RawlsDA
               // between the time we retrieved them and here, where we update them.
               val allOperations = generateAttributeOperations(workspaceResponse.workspace.attributes, userAttrs,
                 k => k.namespace == AttributeName.libraryNamespace && k.name != LibraryService.publishedFlag.name)
-              rawlsDAO.patchWorkspaceAttributes(ns, name, allOperations) map (RequestComplete(_))
+              val republish = workspaceResponse.workspace.attributes.contains(publishedFlag) &&
+                workspaceResponse.workspace.attributes.get(publishedFlag).fold(false)(_.asInstanceOf[AttributeBoolean].value)
```

Comment:
the `contains` part of this clause is redundant - the `attributes.get(publishedFlag) ...` handles the case where the attribute is missing and properly folds it to false.

optional feedback: in general, the codebase uses `getOrElse` instead of `fold`. I don't think one it's necessarily better, but it's more consistent. That said, I think this fold syntax is nice and terse and I'm good with it.

This line will have a problem if the library:published attribute is something other than a boolean. Do we care?